scheduler/core/components/optimizer/greedymax.py

```diff
@@ -403,12 +418,14 @@ def add(self, group_data: GroupData, night: int, interval: Optional[Interval] =
                 # as in calculated somewhere else or in a different way, but are needed when plan.add is called.
                 # In the future we could merge this with timeline but the design on that is TBD.
                 # TODO: Partner calibrations should not contribute to this
-                visit_score = np.sum(
-                    group_data.group_info.scores[night][start_time_slot:start_time_slot + obs_len]
-                )
+                visit_score = sum(group_data.group_info.scores[night][start_time_slot:start_time_slot + obs_len])
 
-                self.obs_in_plan[observation.id] = {'obs': observation, 'obs_len': obs_len,
-                                                    'start': start, 'visit_score': visit_score}
+                self.obs_in_plan[observation.id] = ObsPlanData(
+                    obs=observation,
+                    obs_start=start,
+                    obs_len=obs_len,
+                    visit_score=visit_score
+                )
```

Comment:
With the `ObsPlanData`, I was just trying to get rid of the dictionary on the left here. ☹️ 
Since this is using `Timeline`, I don't think we have enough information yet for a `Visit` yet? I'm going to have to go over the code in finer detail, but I'm anxious to change it too dramatically until Bryan has it running correctly. I think we could manage to combine some methods and come up with something that represents `Timeline` and `Plan` so that this isn't duplicated, but right now, that may not be the best idea. 😦